lib/rouge/lexers/email.rb

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Email < RegexLexer
+      tag 'email'
+      aliases 'eml', 'mail'
+      filenames '*.eml', '*.mail'
+      mimetypes 'message/rfc822'
+
+      title "Email"
+      desc "An email message"
+
+      def self.detect?(text)
+        return true if text.start_with?('From: ')
+      end
+
+      state :root do
+        rule %r/^(From|To|Cc|Bcc):\s/, Keyword, :address
+        rule %r/^Date:\s/, Keyword, :date
+        rule %r/^Subject:\s/, Keyword, :subject
+        rule %r/\n/m, Text::Whitespace
+        rule %r/^>.*/, Comment
+        rule %r/^--\s\n/m, Comment::Doc, :signature
+        rule %r/.*/, Text
```

Comment:
I'd make this `.+$` to a) emphasize that it matches to the end of a line (given no `/m`) but also to make sure it doesn't match the empty string (which can cause performance degradation).

Reply:
Changed; thanks.